### PR TITLE
Release 6.0.0

### PR DESCRIPTION
**Pull request by msilvagarcia**
This releases version 6.0.0 of the Adyen PHP API Library

Remarkable features include:
- Add /getCostEstimate Resource model (#144)
- PW-1996 Remove escaping from notification HMAC (#165)
- Update to latest API version (#162)
- PW-2059 Mask encrypted/sensitive data in the logs (#164)
- Update notification endpoint version (#160)

Because of the update to the latest API version (#162), this is a major release. There are changes to the MarketPay configuration API responses that users need to address after updating.

### CODE REVIEW

File: src/Adyen/HttpClient/CurlClient.php [hunk 1]
```diff
@@ -1,57 +1,92 @@
...
+    // the request array
+
+    private static $requestParamsToMask = array(
+        'paymentData',
+        'card' => array(
+            'number',
+            'cvc'
+        ),
+        'additionalData' => array(
+            'card.encrypted.json',
```
Comment (Aleffio) [inline src/Adyen/HttpClient/CurlClient.php]:
```suggestion
            'card.encrypted.json'
```

File: src/Adyen/Service/ResourceModel/BinLookup/GetCostEstimate.php [hunk 2]
```diff
@@ -0,0 +1,17 @@
+<?php
+
+namespace Adyen\Service\ResourceModel\BinLookup;
+
+class GetCostEstimate extends \Adyen\Service\AbstractResource
+{
```
Comment (Aleffio) [inline src/Adyen/Service/ResourceModel/BinLookup/GetCostEstimate.php]:
```suggestion
{
    /**
     * @var string
     */
    protected $endpoint;
    
    
    /**
     * Include applicationInfo key in the request parameters
     *
     * @var bool
     */
    protected $allowApplicationInfo = false;
    
```

Let's do like the other services?

File: src/Adyen/Service/ResourceModel/BinLookup/GetCostEstimate.php [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    /**
+     * GetCostEstimate constructor.
+     *
+     * @param \Adyen\Service $service
+     */
+    public function __construct($service)
+    {
+        $endpoint = $service->getClient()->getConfig()->get('endpoint') . '/pal/servlet/BinLookup/' . $service->getClient()->getApiBinLookupVersion() . '/getCostEstimate';
+        parent::__construct($service, $endpoint, false);
```
Comment (Aleffio) [inline src/Adyen/Service/ResourceModel/BinLookup/GetCostEstimate.php]:
```suggestion
        parent::__construct($service, $this->endpoint, $this->allowApplicationInfo);
```

File: src/Adyen/Service/ResourceModel/BinLookup/GetCostEstimate.php [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+class GetCostEstimate extends \Adyen\Service\AbstractResource
+{
+    /**
+     * GetCostEstimate constructor.
+     *
+     * @param \Adyen\Service $service
+     */
+    public function __construct($service)
+    {
+        $endpoint = $service->getClient()->getConfig()->get('endpoint') . '/pal/servlet/BinLookup/' . $service->getClient()->getApiBinLookupVersion() . '/getCostEstimate';
```
Comment (Aleffio) [inline src/Adyen/Service/ResourceModel/BinLookup/GetCostEstimate.php]:
```suggestion
        $this->endpoint = $service->getClient()->getConfig()->get('endpoint') . '/pal/servlet/BinLookup/' . $service->getClient()->getApiBinLookupVersion() . '/getCostEstimate';
```

**Comment by msilvagarcia** [comment]
@Aleffio while I believe your suggestions are great, I don't think this Pull Request is the best place to apply them. This would mean we're pushing commits to the develop branch, which should only be done with Pull Requests.

Do you think I should open a new pull request for these changes?

**Comment by Aleffio** [comment]
@msilvagarcia I agree this is not the best place, though this review is like a "second chance" of seeing something before it goes to the master branch. I don't believe these suggestions are so complex to need a separate pull request, but feel free to open one if you think it is needed

**Pull request merged**